src/Swarm/TUI/Model/Failure.hs

```diff
@@ -13,39 +13,26 @@
 -- to create common infrastructure for logging.
 module Swarm.TUI.Model.Failure where
 
-import Data.Char (toLower)
 import Data.Text (Text)
-import Data.Text qualified as T
-import Data.Yaml (ParseException, prettyPrintParseException)
+import Data.Yaml (ParseException)
 
 data SystemFailure
-  = AssetNotLoaded Asset FilePath LoadingFailure
+  = AssetNotLoaded Asset PathLoadFailure
 
-data Asset = Achievement | Data | History | Save
+data PathLoadFailure
+  = PathLoadFailure FilePath LoadingFailure
```

Comment:
Why do you curry error constructor? :slightly_frowning_face: 

As far as I see, you do not use the `PathLoadFailure` for anything else. The `SystemFailure` in question is already named `AssetNotLoaded` so what do you get by introducing this extra step?

Reply:
The `SystemFailure` type has an `Asset` member, and it was unclear [in certain low-level functions](https://github.com/swarm-game/swarm/pull/1116/files#r1111098032) which asset is relevant.  However, in the higher-level functions, the context becomes clear, and then I am able to use `withExceptT` to [supplement the error with the `Asset` argument](https://github.com/swarm-game/swarm/pull/1116/files#r1111100570).

Reply:
@kostmo I only see the `getData[Dir|File]Safe` functions. Why do this hard, when you can simply pass them the `Asset` type?
```diff
- left (AssetNotLoaded $ Data AppAsset)  <$> getDataDirSafe "."
+ getDataDirSafe AppAsset "."
```

This also makes more sense, because no other `Asset` is in the `data` directory.